crates/codegen/syntax_templates/src/rust/parser_output.rs

```diff
@@ -27,13 +27,25 @@ impl ParseOutput {
 
 #[derive(PartialEq)]
 pub struct ParseError {
-    pub(crate) position: TextPosition,
+    pub(crate) range: TextRange,
     pub(crate) expected: BTreeSet<String>,
 }
 
 impl ParseError {
-    pub fn position(&self) -> &TextPosition {
-        return &self.position;
+    pub fn range(&self) -> &TextRange {
+        return &self.range;
+    }
+
+    pub fn utf8_range(&self) -> Range<usize> {
```

Comment:
suggestion: can we implement these utilities for `TextRange` instead, so they can be used everywhere else, not just `ParseError`? A few places in the codebase already do that. I imagine other Rust clients will also do that conversion.

```rust
trait TextRangeExtensions {
  fn utf8(&self) -> Range<usize>;
  fn utf16(&self) -> Range<usize>;
  fn chars(&self) -> Range<usize>;
}

impl TextRangeExtensions for TextRange {
  ...
}
```